commands/evergreen.js

```diff
@@ -39,20 +37,45 @@ module.exports = function (input, flags, options) {
       }))
         .then((existingFiles) => {
           if (existingFiles.length && !flags.force) {
-            console.error('Error: evergreen update ZIPs already exist in www/');
-            console.error('Please remove them or use --force to overwrite');
-            return process.exit(1);
+            return Promise.reject(new Error(`Evergreen update ZIPs already exist in www/
+Please remove them or use --force to overwrite`));
           }
           return platforms;
         });
     })
     .then((platforms) => {
-      // serially create evergreen update ZIPs
-      console.log('creating evergreen update ZIPs...');
+      // serially create evergreen update ZIPs and upload if required
+      console.log('Creating evergreen update ZIPs...');
       // (prev: Promise, platform: String) => Promise
       return platforms.reduce((prev, platform) => {
         return prev.then(() => evergreen.zipPlatform(projectPath, platform))
-          .then((zipPath) => console.log(`created: ${zipPath}`));
+          .then((zipPath) => {
+            console.log(`Created: ${zipPath}`);
+            if (!flags.upload) {
+              return zipPath;
+            }
+            return Promise.all([
+              evergreen.getCurrentTenant(options.blinkMobileIdentity),
+              cordovaProjectId(projectPath),
+              evergreen.assumeAWSRole(options.blinkMobileIdentity)
+            ])
+              .then((results) => evergreen.upload(zipPath, results[0], results[1], results[2]))
+              .then((location) => console.log(`Remote location: ${location}`))
```

Comment:
Yeah it this doesnt need to return anything as the reduce just passes the promise to next platform and the zip/upload doesnt need any input